### ISSUE TITLE
Spotify: add offset parameter

### ISSUE THREAD

**Issue by tennox** [n8n-io/n8n#2554]
**Describe the bug**
It is currently impossible (afaik) to fetch entries >1000 on spotify - e.g. liked tracks, as the limit parameter is maxed at 1000 from Spotify API side.

So it's needed to be able to specify `offset` parameter.
This will also allow to process paginated batches.
 
**To Reproduce**
1. Fetch "liked tracks" when you have >1000
2. Receive only 1000 items

**Expected behavior**
Receive all items (when you select "return all") and/or have a possibility to do proper pagination with offset Parameter.

**Additional context**
Willing to submit a PR when there's an agreement on how to handle the situation.

**Comment by Joffcom** [comment]
Hey @tennox,

It looks like the `spotifyApiRequestAllItems()` function is hard coded to only work up to 1000 entries in batches of 20, Not sure why it is done like this @RicardoE105 may know as he put in a fix for Pagination a few weeks back.

A quick fix is to just remove the lines below from the `GenericFunctions.ts` file, In a quick test here that has worked.

```
if (uri?.includes('offset=1050')) {
    return returnData;
}
```

**Comment by Joffcom** [comment]
Was about to pop in a pull request and I found another issue, Looks like Get New Releases doesn't work.

**Comment by Joffcom** [comment]
Popped a pull request in to fix the pagination issue and the get new album releases operation.

Now to clear up my liked tracks :joy:

**Comment by janober** [comment]
@RicardoE105 Do you have any idea why that got added?

**Comment by YDeltagon** [comment]
hello, Have you any news ?
I'm also stuck at 1000, is there a way around it?

**Comment by YDeltagon** [comment]
I don't find the "GenericFunctions.ts" on the Windows desktop version

**Comment by Joffcom** [comment]
Hey @YDeltagon,

The desktop version is a bit different so you wouldn't be able to compile up your own change. You might be able to manually patch the files if you can find them though.

**Comment by janober** [comment]
Issue got fixed with [this](https://github.com/n8n-io/n8n/commit/4c029e54010eb5abc35ffd8a5897fc58f9a1f1e8) commit.

The limit was in there because the '/search' endpoint has the 1k limit. 

Will be released with the next version.

**Comment by janober** [comment]
Got released with `n8n@0.160.0`

**Comment by YDeltagon** [comment]
Nice, I don't know yet how to update the desktop version though :D
You'r fast :D
